### PR TITLE
[doc] fix amp doc

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125673

# Motivation
The doc shown on https://pytorch.org/docs/main/amp.html#torch.amp.autocast_mode.is_autocast_available should be `torch.amp.is_autocast_available` instead of `torch.amp.autocast_mode.is_autocast_available`. This PR aims to fix it.

cc @mcarilli @ptrblck @leslie-fang-intel @jgong5

### CODE REVIEW

**Comment by guangyey** [comment]
@albanD Sorry, I made a mistake on the document `is_autocast_available`. This PR aims to fix it.

File: docs/source/conf.py [hunk 1]
```diff
@@ -159,6 +159,7 @@
     "hierarchical_pickle",
     # torch.amp.autocast_mode
     "autocast_decorator",
+    "is_autocast_available",
```
[the remark this answers is not in the document]
Comment (guangyey) [reply docs/source/conf.py:162]: OK, let me check.

**Comment by albanD** [comment]
CI seems to be not so happy right now?

**Comment by guangyey** [comment]
> CI seems to be not so happy right now?

I can't fully understand the syntax of these docs, so I'm experimenting with them over and over again :(

**Review by albanD — COMMENTED** [review]
Yes they are a bit finicky. Especially for re-exposed APIs.

**Comment by guangyey** [comment]
Close this PR as it is expected.

**Pull request closed without merging**